Updates documentation update (#962)
bottlerocket-os/bottlerocket: README.md

aws ssm start-session --target INSTANCE_ID
```

> With the [default control container](https://github.com/bottlerocket-os/bottlerocket-control-container), you can make API calls to change settings and/or update your Bottlerocket host to different image versions.

Reviewer: ```suggestion
With the [default control container](https://github.com/bottlerocket-os/bottlerocket-control-container), you can make [API calls](#api) to configure and manage your Bottlerocket host.
```
I think we can go a little more general here so we're not updating it each time the API can do something new.

Rather than a package manager that updates individual pieces of software, Bottlerocket downloads a full filesystem image and reboots into it.
It can automatically roll back if boot failures occur, and workload failures can trigger manual rollbacks.

The update process uses images secured by [TUF](https://theupdateframework.github.io/).
For more details, see the [update system documentation](sources/updater/).

> There are several ways of updating your Bottlerocket instances:

Reviewer: ```suggestion
### Update methods

There are several ways of updating your Bottlerocket instances.
```
Don't want to jump up by multiple heading levels.

#### Update API

The [Bottlerocket API](#api) allows you to update the instance from within the [control container](#control-container) or any [host container](#custom-host-containers) that has access to the Bottlerocket API.

The update API is made up of a set of actions endpoints and a status endpoint to check the result of those actions.
The behavior of the update API is governed by the [update settings](#updates-settings).

In general, the update process looks like the following:

Refresh list of available updates:
```
apiclient -u /actions/refresh-updates -m POST
```

Then check the list of available updates with:
```

> apiclient -u /updates/status

Reviewer 2: Maybe it would be beneficial to show the output of this command and explain how an update is the chosen_update?

Author: This section mainly serves as a high level overview of the steps to do a basic update. I'm somewhat reluctant to include the full structure of the update status here as I fear that might distract users from the commands to actually run. Following the steps, the user should be able to directly see the output and the referenced `chosen_update` field.

As for how an update is the `chosen_update`, that's determined by what is in the update settings which I reference a few lines above. I'll go ahead and add a reference to the `version-lock` setting to make it more clear.

```

If the `chosen_update` looks right, you can prepare said update with:
```
apiclient -u /actions/prepare-update -m POST
```

After update gets being downloaded and applied, you can check the update status again with:
```
apiclient -u /updates/status
```

If the `staging_partition` image information looks right (should be the chosen update), you can proceed to "activate" the update with:
```
apiclient -u /actions/activate-update -m POST
```

After activating the update, the next reboot will boot into this new chosen version. You can reboot the host with:
```
apiclient -u /actions/reboot -m POST
```

#### Updog

> You can update using a CLI tool, updog.

Reviewer: ```suggestion
You can also update using a CLI tool, `updog`, if you [connect through a host container](#exploration).
```

Rather than a package manager that updates individual pieces of software, Bottlerocket downloads a full filesystem image and reboots into it.
It can automatically roll back if boot failures occur, and workload failures can trigger manual rollbacks.

The update process uses images secured by [TUF](https://theupdateframework.github.io/).
For more details, see the [update system documentation](sources/updater/).

There are several ways of updating your Bottlerocket instances:

#### Update API

The [Bottlerocket API](#api) allows you to update the instance from within the [control container](#control-container) or any [host container](#custom-host-containers) that has access to the Bottlerocket API.

The update API is made up of a set of actions endpoints and a status endpoint to check the result of those actions.

> The behavior of the update API is governed by the [update settings](#updates-settings).

Reviewer: ```suggestion
The [Bottlerocket API](#api) allows you to update and reboot your host with simple API calls.  You can change [settings](#updates-settings) to control which updates are selected.
```
* "instance" -> "host" to not be too specific
* Remove container mentions since the customer could transport API calls in other ways, and we describe API access methods elsewhere
* Overall simplification, since we're about to show the reader the endpoints.

In general, the update process looks like the following:

> Refresh list of available updates:

Reviewer: ```suggestion
First, refresh the list of available updates:
```

```
apiclient -u /actions/refresh-updates -m POST
```

> Then check the list of available updates with:

Reviewer: ```suggestion
Now you can see the list of available updates, along with the chosen update, according to your `version-lock` [setting](#updates-settings):
```

```
apiclient -u /updates/status
```

> This will return the current update status in JSON format. For example, the status should look something like the following (pretty-printed):

Reviewer: ```suggestion
This will return the current update status in JSON format. The status should look something like the following (pretty-printed):
```

```
{
  "update_state": "Available",
  "available_updates": [
    "0.4.0",
    "0.3.4",
    ...
  ],
  "chosen_update": {
    "arch": "x86_64",
    "version": "0.4.0",
    "variant": "aws-k8s-1.15"
  },
  "active_partition": {
    "image": {
      "arch": "x86_64",
      "version": "0.3.2",
      "variant": "aws-k8s-1.15"
    },
    "next_to_boot": true
  },
  "staging_partition": null,
  "most_recent_command": {
    "cmd_type": "refresh",
    "cmd_status": "Success",
    ...
  }
}
```

> If the `chosen_update` (see `settings.updates.version-lock` in [update settings](#updates-settings)) looks right, you can prepare said update with:

Reviewer: ```suggestion
You can see that the we're running `v0.3.2` in the active partition, and that `v0.4.0` is available.  If you're happy with that selection, you can request that the update be downloaded and applied to disk.  (The update will remain inactive until you make the `activate-update` call below.)
```

```
apiclient -u /actions/prepare-update -m POST
```

> After update gets being downloaded and applied, you can check the update status again with:

Reviewer: ```suggestion
After you request that the update be prepared, you can check the update status again until it reflects the new version in the staging partition.
```

```
apiclient -u /updates/status
```

> If the `staging_partition` image information looks right (should be the chosen update), you can proceed to "activate" the update with:

Reviewer: ```suggestion
If the staging partition shows the new version, you can proceed to "activate" the update.  This means that as soon as the host is rebooted it will try to run the new version.  (If the new version can't boot, we automatically flip back to the old version.)
```

```
apiclient -u /actions/activate-update -m POST
```

> After activating the update, the next reboot will boot into this new chosen version. You can reboot the host with:

Reviewer: ```suggestion
You can reboot the host with:
```

Rather than a package manager that updates individual pieces of software, Bottlerocket downloads a full filesystem image and reboots into it.
It can automatically roll back if boot failures occur, and workload failures can trigger manual rollbacks.

The update process uses images secured by [TUF](https://theupdateframework.github.io/).
For more details, see the [update system documentation](sources/updater/).

### Update methods

There are several ways of updating your Bottlerocket hosts:

#### Update API

The [Bottlerocket API](#api) allows you to update and reboot your host with simple API calls.  You can change [settings](#updates-settings) to control which updates are selected.

> In general, the update process looks like the following:
> 
> First, refresh list of available updates:

Reviewer: ```suggestion
First, refresh the list of available updates:
```

Rather than a package manager that updates individual pieces of software, Bottlerocket downloads a full filesystem image and reboots into it.
It can automatically roll back if boot failures occur, and workload failures can trigger manual rollbacks.

The update process uses images secured by [TUF](https://theupdateframework.github.io/).
For more details, see the [update system documentation](sources/updater/).

### Update methods

There are several ways of updating your Bottlerocket hosts:

#### Update API

The [Bottlerocket API](#api) allows you to update and reboot your host with simple API calls.  You can change [settings](#updates-settings) to control which updates are selected.

> In general, the update process looks like the following:

Reviewer: ```suggestion
In general, the process of using the update API looks like this.  You refresh the list of known updates, then apply one to the system.  Calls to `/updates/status` will tell you the current state and give more details on any errors.
```

> ![Update API overview](sources/api/Update%20API.png)

Reviewer: Can we change the space in the name to a hyphen or something?  Maybe lowercase them?  Sorry!
